fs/fs.go

```diff
@@ -346,6 +346,19 @@ func (fs *filesystem) MountLocal(ctx context.Context, mountpoint string, labels
 	}
 	unpacker := NewLayerUnpacker(fetcher, archive)
 	desc := s.Target
+
+	// If no descriptor size is given, resolve the layer
+	// to populate it
+	if desc.Size == 0 {
+		// In remoteStore.Reference, Registry and Target will always be correct.
+		// However, we need to update Reference to point to the correct digest
+		remoteStore.Reference.Reference = s.Target.Digest.String()
+		desc, err = remoteStore.Blobs().Resolve(ctx, remoteStore.Reference.String())
+		if err != nil {
+			return fmt.Errorf("cannot resolve local blobs %s: %w", remoteStore.Reference.String(), err)
```

Comment:
nit: I'm not sure this message is clear without knowing the implementation details. What about:
```
"cannot resolve size of layer %s: %w"
```

Reply:
Agreed. What ever it is, I think we should keep the error format consistent with https://github.com/awslabs/soci-snapshotter/blob/95c139e65a4bdfca058cec324403afad0f8fcd58/fs/artifact_fetcher.go#L121.